client/src/scripts/Socket.js

```diff
@@ -0,0 +1,68 @@
+import {io} from "socket.io-client";
+import GameManager from "./GameManager";
+import MapCreator from "./MapCreator";
+
+export default class Socket {
+    static instance = null;
+
+    constructor(room) {
+        Socket.instance = this;
```

Comment:
check if instance != null then return null (so you dont overrite instance by mistake)